dask/array/core.py

```diff
@@ -2649,15 +2682,48 @@ def elemwise(op, *args, **kwargs):
                                                   tokenize(op, dt, *args))
 
     if other:
-        return atop(partial_by_order, expr_inds,
+        result = atop(partial_by_order, expr_inds,
                     *concat((a, tuple(range(a.ndim)[::-1])) for a in arrays),
                     dtype=dt, name=name, function=op, other=other,
                     token=funcname(op))
     else:
-        return atop(op, expr_inds,
+        result = atop(op, expr_inds,
                     *concat((a, tuple(range(a.ndim)[::-1])) for a in arrays),
                     dtype=dt, name=name)
 
+    return handle_out(out, result)
+
+
+def handle_out(out, result):
```

Comment:
This *maybe* could be implemented as a decorator, when functions have multiple return points (though I see you changed the one case where that happened). If `elemwise` was the only one that needed to be changed, then a regular function seems preferable.